src/cdk-experimental/popover-edit/form-value-container.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export interface Entry<FormValue> {
+  value?: FormValue;
+}
+
+export class FormValueContainer<Key extends object, FormValue> {
+  private _formValues = new WeakMap<Key, Entry<FormValue>>();
+
+  for(key: Key): Entry<FormValue> {
+    const _formValues = this._formValues;
+
+    let entry = _formValues.get(key);
+    if (!entry) {
+      entry = {};
```

Comment:
I'm not actually super familiar with the forms stuff- why is it beneficial to always default to `{}`?

Reply:
It's so that two-way binding can work without having to do the [] and () parts as separate expressions, not anything forms-specific.

From the example:
[(matEditLensPreservedFormValue)]="weightValues.for(element).value"

Reply:
Could you add comments that explain that?